mq/inmemory/mq.h

```diff
@@ -2,30 +2,104 @@
 #define BRICKS_MQ_INMEMORY_MQ_H
 
 // MMQ is an efficient in-memory FIFO buffer.
+// One of the objectives of MMQ is to minimize the time for which the message pushing thread is blocked for.
 //
-// Messages can be pushed into it via `PushMessage()`.
+// Messages can be pushed into it via thread-safe methods `PushMessage()` or `EmplaceMessage()`.
 // The consumer is run in a separate thread, and is fed one message at a time via `OnMessage()`.
-// The order of messages is preserved.
```

Comment:
See it down there, makes sense. Still needs a clarification comment though, here or below, that except for the multithreading lock case, the order of messages is preserved, and, specifically, the order of messages inserted from the same thread is always kept.